contracts/hooks/examples/FullRange.sol

```diff
@@ -0,0 +1,326 @@
+            afterSwap: false,
+            beforeDonate: false,
+            afterDonate: false
+        });
+    }
+
+    function addLiquidity(
+        address tokenA,
+        address tokenB,
+        uint24 fee,
+        uint256 amountADesired,
+        uint256 amountBDesired,
+        address to,
+        uint256 deadline
+    ) external ensure(deadline) returns (uint128 liquidity) {
+        require(amountADesired > 1000 && amountBDesired > 1000, "Input amount does not minimum liquidity");
+
+        PoolKey memory key = PoolKey({
+            currency0: Currency.wrap(tokenA),
+            currency1: Currency.wrap(tokenB),
+            fee: fee,
+            tickSpacing: 60,
+            hooks: IHooks(address(this))
+        });
+
+        PoolId poolId = key.toId();
+
+        (uint160 sqrtPriceX96,,,,,) = poolManager.getSlot0(poolId);
+
+        if (sqrtPriceX96 == 0) revert PoolNotInitialized();
+
+        PoolInfo storage pool = poolInfo[poolId];
+
+        uint128 poolLiquidity = poolManager.getLiquidity(poolId);
+
+        liquidity = LiquidityAmounts.getLiquidityForAmounts(
+            sqrtPriceX96,
+            TickMath.getSqrtRatioAtTick(MIN_TICK),
+            TickMath.getSqrtRatioAtTick(MAX_TICK),
+            amountADesired,
+            amountBDesired
+        );
+
+        // require(amountADesired > 1000 && amountBDesired > 1000, "Input amount does not minimum liquidity");
+
+        modifyPosition(
+            key,
+            IPoolManager.ModifyPositionParams({
+                tickLower: MIN_TICK,
+                tickUpper: MAX_TICK,
+                liquidityDelta: int256(int128(liquidity))
+            })
+        );
+
+        if (poolLiquidity == 0) {
+            liquidity -= MINIMUM_LIQUIDITY;
+            UniswapV4ERC20(pool.liquidityToken).mint(address(0), MINIMUM_LIQUIDITY); // permanently lock the first MINIMUM_LIQUIDITY tokens
+        }
+
+        UniswapV4ERC20(pool.liquidityToken).mint(to, liquidity);
+    }
+
+    function removeLiquidity(address tokenA, address tokenB, uint24 fee, uint256 liquidity, uint256 deadline)
+        public
+        virtual
+        ensure(deadline)
+        returns (BalanceDelta delta)
+    {
+        PoolKey memory key = PoolKey({
+            currency0: Currency.wrap(tokenA),
+            currency1: Currency.wrap(tokenB),
+            fee: fee,
+            tickSpacing: 60,
+            hooks: IHooks(address(this))
+        });
+
+        PoolId poolId = key.toId();
+
+        (uint160 sqrtPriceX96,,,,,) = poolManager.getSlot0(poolId);
+
+        if (sqrtPriceX96 == 0) revert PoolNotInitialized();
+
+        UniswapV4ERC20 erc20 = UniswapV4ERC20(poolInfo[poolId].liquidityToken);
+
+        erc20.burn(msg.sender, liquidity);
+
+        delta = modifyPosition(
+            key,
+            IPoolManager.ModifyPositionParams({
+                tickLower: MIN_TICK,
+                tickUpper: MAX_TICK,
+                liquidityDelta: -int256(liquidity)
```

Comment:
when we call the actual poolManager liquidity will have increased bc of accrued fees, right? So when we burn we need to get `(liquidity / totalSupply) * manager.getLiquidity(pool)` basically a % of all liquidity owned, but take that percentage out of the actual pool's liquidity.

Reply:
(maybe this is why liquidity is getting weird in some of your fuzz tests??)